plugins/outputs/iotdb/iotdb.go

```diff
@@ -0,0 +1,277 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package iotdb
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/apache/iotdb-client-go/client"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//go:embed sample.conf
+var sampleConfig string
+
+type IoTDB struct {
+	Host            string          `toml:"host"`
+	Port            string          `toml:"port"`
+	User            string          `toml:"user"`
+	Password        string          `toml:"password"`
+	Timeout         config.Duration `toml:"timeout"`
+	ConvertUint64To string          `toml:"uint64_conversion"`
+	TimeStampUnit   string          `toml:"timestamp_precision"`
+	TreatTagsAs     string          `toml:"convert_tags_to"`
+
+	session *client.Session
+	Log     telegraf.Logger `toml:"-"`
+}
+
+type recordsWithTags struct {
+	// IoTDB Records basic data struct
+	DeviceIDList     []string
+	MeasurementsList [][]string
+	ValuesList       [][]interface{}
+	DataTypesList    [][]client.TSDataType
+	TimestampList    []int64
+	// extra tags
+	TagsList [][]*telegraf.Tag
+}
+
+func (*IoTDB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init is for setup, and validating config.
+func (s *IoTDB) Init() error {
+	if time.Duration(s.Timeout)/time.Millisecond < 0 {
+		return errors.New("negative timeout")
+	}
+	if !choice.Contains(s.ConvertUint64To, []string{"int64", "int64_clip", "text"}) {
+		return fmt.Errorf("unknown 'uint64_conversion' method %q", s.ConvertUint64To)
+	}
+	if !choice.Contains(s.TimeStampUnit, []string{"second", "millisecond", "microsecond", "nanosecond"}) {
+		return fmt.Errorf("unknown 'timestamp_precision' method %q", s.TimeStampUnit)
+	}
+	if !choice.Contains(s.TreatTagsAs, []string{"fields", "device_id"}) {
+		return fmt.Errorf("unknown 'convert_tags_to' method %q", s.TreatTagsAs)
+	}
+	s.Log.Info("IoTDB output plugin initialization completed.")
+	return nil
+}
+
+func (s *IoTDB) Connect() error {
+	sessionConf := &client.Config{
+		Host:     s.Host,
+		Port:     s.Port,
+		UserName: s.User,
+		Password: s.Password,
+	}
+	var ss = client.NewSession(sessionConf)
+	s.session = &ss
```

Comment:
```suggestion
	s.session = &client.NewSession(sessionConf)
```

Reply:
I tried to change it like this, but it raised an error. So, I didn't change it in my last commit. 
This change:
```
s.session = &(client.NewSession(sessionConf))
```
also raised an error.
Maybe it's because `s.session` is a pointer but can't pointer directly to a return value? 
Or how about:
```
ss := client.NewSession(sessionConf)
s.session = &ss
```
I'm not so familiar with Golang, so please feel free to point out my mistakes in time. And thanks for your patient guidance.

Reply:
Ah yes I see... Keep it then...